### PR TITLE
[clang-tidy] Let `bugprone-use-after-move` also handle calls to `std::forward`

### LINKED ISSUE #82023: clang-tidy bugprone-use-after-move check does not catch misuse of std::forward<>()
The clang-tidy bugprone-use-after-move check does not also evaluate perfect forwarding, which has similar risks.  Example:
```
#include <utility>

struct S {
  S();
  S(S&&);
};

void consume(S s);

template <typename T>
void forward(T&& t) {
  consume(std::forward<T>(t));
  consume(std::forward<T>(t));  // use-after-forward
}

void create() {
  S s;
  forward(std::move(s));
}
```
Live repro: https://godbolt.org/z/M3qWeW8Gh

Note that the clang-analyzer-cplusplus.Move check would have caught this case. (I don't know why there are two distinct use-after-move checks. Should one be turned down in favor of the other?)

### PR DESCRIPTION
**Pull request by AMS21**
Fixes #82023

### CODE REVIEW

**Comment by 5chmidti** [comment]
I'd like to see the diagnostics say `forwarded` instead of `moved` (for all 4 possible diagnostics). `after it  was potentially moved by forwarding` could work as well. Otherwise, the changes themselves look good

**Comment by AMS21** [comment]
Rebased and address review comment

File: clang-tools-extra/clang-tidy/bugprone/UseAfterMoveCheck.cpp [hunk 1]
```diff
@@ -359,24 +360,52 @@ void UseAfterMoveFinder::getReinits(
...
-                DiagnosticIDs::Note);
+                DiagnosticIDs::Note)
+        << ActionType;
   } else if (UseLoc < MoveLoc || Use.DeclRef == MoveArg) {
-    Check->diag(UseLoc,
-                "the use happens in a later loop iteration than the move",
-                DiagnosticIDs::Note);
+    Check->diag(UseLoc, "the use happens in a later loop iteration than the %0",
+                DiagnosticIDs::Note)
+        << ActionType;
```
Comment (5chmidti) [inline clang-tools-extra/clang-tidy/bugprone/UseAfterMoveCheck.cpp:408]: You could use the diagnostics 'select' syntax to do this in a more compact way: https://github.com/llvm/llvm-project/blob/be083dba95dfbbb0286d798cc06fbe021715bc03/clang-tools-extra/clang-tidy/hicpp/SignedBitwiseCheck.cpp#L96-L98

File: clang-tools-extra/clang-tidy/bugprone/UseAfterMoveCheck.cpp [hunk 2]
```diff
@@ -359,24 +360,52 @@ void UseAfterMoveFinder::getReinits(
...
+  Forward, // std::forward
+};
+
+static MoveType determineMoveType(const FunctionDecl *FuncDecl) {
+  if (FuncDecl->getName() == "move")
+    return MoveType::Move;
+  if (FuncDecl->getName() == "forward")
+    return MoveType::Forward;
+
+  assert(false && "Invalid move type");
```
Comment (5chmidti) [inline clang-tools-extra/clang-tidy/bugprone/UseAfterMoveCheck.cpp:374]: Use `llvm_unreachable("Invalid move type")` if this code should truly be unreachable (which it is from what I can tell).

**Review by 5chmidti — COMMENTED** [review]
Please add a short explanation that `forward` is also checked in the check's docs here https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/docs/clang-tidy/checks/bugprone/use-after-move.rst.

**Comment by AMS21** [comment]
Rebased and implemented review suggestions

**Review by PiotrZSL — APPROVED** [review]
Overall, LGTM

**Review by 5chmidti — APPROVED** [review]

**Review by HerrCai0907 — APPROVED** [review]
LGTM

**Pull request merged**